### DISCUSSION TITLE
Why was the TeamsFX AzFunc Extension removed from the templates?

### DISCUSSION THREAD

**Discussion by andrewconnell** [OfficeDev/microsoft-365-agents-toolkit#12013] [Q&A]
Previously (late 2023), the Azure Function included in the **Tab > React with Fluent UI** template included the TeamsFx Azure Function extension... but in the latest templates for TypeScript, it's gone. Why? It's no longer using the **TeamsFxContext** input binding that was there (and was so useful).

The Azure Function in the template seems to have outdated documentation (_looks like it still references Azure Functions v3 when it's clear it's switched to the v4 format_) so it's hard to understand how to add the binding, but all the environment variables (`M365_CLIENT_ID`, `M365_CLIENT_SECRET`, `M365_TENANT_ID`, `M365_AUTHORITY_HOST`, `ALLOWED_APP_IDS`) used by the extension and binding are still added to the config... so are they still used if the extension isn't installed?

This is very confusing... it looks like the Azure Function has been sort of upgraded, but the job wasn't finished and the extension was intended to be removed, but things weren't fully updated...

**Comment by blackchoey** [comment]
@andrewconnell The major functionality in the TeamsFx Azure Function extension has be consolidated into latest Azure Function. So you no longer need to include the binding in your project, which also reduces the requirement of dotnet during development.
The environment variables are also used in the Azure Function's business logic in the template, so it's still required to be configured. Only `ALLOWED_APP_IDS` is not needed any more. We will double confirm and remove that environment variable from the template.

**Reply by andrewconnell** [reply]
Interesting... but what of the `TeamsFx` input binding that was provided? That was very useful in that it provided a context from where the request originated from in Teams... that seems gone :(

BTW, was there any announcement or mention in the changelog that I was missing when this change was made? This is a significant change that seems to have been made with no announcement... unless I'm missing something.

**Reply by blackchoey** [reply]
The major functionality provided by TeamsFx binding is to provide improved security to your Azure Function API, so you don't need to add extra authorization logic in your API to prevent unauthorized access. Now the Azure Function service provides built-in support for it and the template is updated to use the Azure Function service's built-in feature. The functionality of the template itself does not change.

For ` it provided a context from where the request originated from in Teams`, can you explain more? The only information in the context is the access token, which is obtained from request header so you can easily get it with `req.headers.get("Authorization")?.replace("Bearer ", "").trim();` (see the latest template).

You can still use the binding for older Azure Function version if needed. And feel free to open feature request in GitHub issues with your scenario so we can further follow up with it.
